sklearn/_config.py

```diff
@@ -14,6 +14,7 @@
     ),
     "enable_cython_pairwise_dist": True,
     "array_api_dispatch": False,
+    "transform_output": "default",
```

Comment:
This option use to be `"numpy"`, but it was inconsistent with transformers that can output sparse data. I was also thinking about third party transformers that already output dataframes where a "numpy" default would be strange.

I can get behind `"array-like"`. The only concern I have is how sparse data is a weird "array-like", because `asarray` on a sparse matrix returns an object dtype:

```python
from scipy.sparse import csr_matrix
import numpy as np

mat = csr_matrix([[1, 2, 0]])
print(np.asarray(mat).dtype)
# object
```

Reply:
Looking at our [glossary entry](https://scikit-learn.org/stable/glossary.html#term-array-like) for "array-like", we exclude "sparse matrix" from "array-like". In that case, "array-like" would not be a good default, because it does not cover sparse matrices.

Reply:
Do you have a proposal. Naming the default "default" just seems wrong to me. What if we change it in the future?

Reply:
The semantics for "default" is "the transformer does anything it wants". Here are some options:

1. `None`
2. `"undefined"`
3. `"unmodified"`
4. `"unchanged"`

I am in favor of `None`. I think it's the most pythonic way to say "use the default".

Reply:
I'm fine with `None`. What do others think?